### PR TITLE
[NPU] Adding Driver Mapped Inference and ELF format version to Caching Properties

### PR DESCRIPTION
**Pull request by dvpv**
### Details:
 - Replacing Driver Version with Driver Mapped Inference and Driver ELF Format versions for cache invalidation in order to avoid unnecessary cache invalidation.
 - In case the driver does not have MI and ELF Format Versions API, the cache invalidation input will remain as before, driver version only.

**- Need to investigate OV cache invalidation logic to be sure that it will work with different caching properties.**

PR Parked until [PR-24502](https://github.com/openvinotoolkit/openvino/pull/24502) is merged.
 
### Tickets:
  - EISW-101841

### CODE REVIEW

File: src/plugins/intel_npu/src/compiler/src/zero_compiler_in_driver.cpp [hunk 1]
```diff
@@ -507,6 +507,23 @@ std::string LevelZeroCompilerInDriver<TableExtension>::serializeConfig(
         content = std::regex_replace(content, std::regex(batchstr.str()), "");
     }
 
+    // DRIVER_ELF_FROMAT_VERSION and DRIVER_MI_VERSION are not supported in versions < 5.7 - need to remove them
+    if ((compilerVersion.major < 5) || (compilerVersion.major == 5 && compilerVersion.minor < 7)) {
```
Comment (razvanapetroaie) [inline src/plugins/intel_npu/src/compiler/src/zero_compiler_in_driver.cpp:511]: 5.7 is the current version. Shouldn't we use 5.8 as the threshold?

Comment (dvpv) [reply src/plugins/intel_npu/src/compiler/src/zero_compiler_in_driver.cpp]: Fixed

Comment (razvanapetroaie) [reply src/plugins/intel_npu/src/compiler/src/zero_compiler_in_driver.cpp]: 5.9 now

File: src/plugins/intel_npu/src/plugin/include/metrics.hpp [hunk 2]
```diff
@@ -61,7 +61,8 @@ class Metrics final {
                                                               ov::intel_npu::dpu_groups.name(),
                                                               ov::intel_npu::dma_engines.name(),
                                                               ov::intel_npu::compilation_mode.name(),
-                                                              ov::intel_npu::driver_version.name(),
+                                                              ov::intel_npu::driver_mi_version.name(),
```
Comment (dvpv) [inline src/plugins/intel_npu/src/plugin/include/metrics.hpp:64]: Update the list dynamically to cover drivers that don't support versions API.

**Review by sbutnari — APPROVED** [review]

**Review by alexandruenache1111 — APPROVED** [review]

**Comment by dvpv** [comment]
PR Parked 😢

**Pull request closed without merging**